### PR TITLE
Update PasswordView with customized password visibility icon

### PR DESCRIPTION
**Pull request by Henry0422**
To enable the passwordIcon feature when initializing password property:
```
{
   class: 'Password',
   name: 'password',
   view: { class: 'foam.u2.view.PasswordView', passwordIcon: true }
}
```

In a particular view, customize the location of the visibility icon using CSS:
```
^ .input-image {
   position: absolute;
   width: XXpx;
   height: XXpx;
   bottom: XXpx;
   right: XXpx;
}
```

If the view (such as `SignInView` & `ResetView`) needs `full-width-input`, add customized css class`^ .full-width-input-password`

### CODE REVIEW

**Comment by Henry0422** [comment]
@kgrgreer I am not sure how to get the relative path of an icon in foam2.

After I add an icon file under `foam2/src/foam/u2` & put `{"class":"foam.nanos.servlet.ServletMapping","className":"foam.nanos.servlet.ImageServlet","pathSpec":"/images/*",initParameters:{"paths":"foam2/src/foam/u2/images"}}` at the end of the `http` in the  foam2 services file, the web page still cannot load the image based on the relative path (`/images/baseline-visibility_off-24px.svg`)

File: src/foam/nanos/auth/resetPassword/ResetView.js [hunk 1]
```diff
@@ -167,9 +176,9 @@ foam.CLASS({
         .start().addClass('Reset-Password').add("Reset Password").end()
         .start().addClass('Message-Container')
           .start().addClass('newPassword-Text').add("New Password").end()
-          .start(this.NEW_PASSWORD).addClass('full-width-input').end()
+          .start(this.NEW_PASSWORD).end()
           .start().addClass('confirmPassword-Text').add("Confirm Password").end()
-          .start(this.CONFIRM_PASSWORD).addClass('full-width-input').end()
+          .start(this.CONFIRM_PASSWORD).end()
```
Comment (kgrgreer) [inline src/foam/nanos/auth/resetPassword/ResetView.js]: You can just do .add(this.CONFIRM_PASSWORD), which is shorter than start() and end().

File: src/foam/u2/view/PasswordView.js [hunk 2]
```diff
@@ -18,7 +18,85 @@
 foam.CLASS({
   package: 'foam.u2.view',
   name: 'PasswordView',
-  extends: 'foam.u2.tag.Input',
+  extends: 'foam.u2.View',
 
-  properties: [ [ 'type', 'password' ] ]
+  css: `
+    .input-field-container {
```
Comment (kgrgreer) [inline src/foam/u2/view/PasswordView.js]: These are global names. Try not to pollute the global namespace, so use ^ and myClass() instead.

File: src/foam/u2/view/PasswordView.js [hunk 3]
```diff
@@ -18,7 +18,85 @@
...
+    },
+    {
+      class: 'Boolean',
+      name: 'passwordInvisible',
+      value: true
+    },
+    {
+      class: 'Boolean',
+      name: 'passwordIcon',
+      defaultValue: false
```
Comment (kgrgreer) [inline src/foam/u2/view/PasswordView.js]: defaultValue is from FOAM1, use value: instead. However, false is already the default so not needed.

File: src/foam/nanos/auth/SignInView.js [hunk 4]
```diff
@@ -123,7 +132,7 @@ foam.CLASS({
           .start().addClass('label').add("Email Address").end()
           .start(this.EMAIL).addClass('full-width-input').end()
           .start().addClass('label').add("Password").end()
-          .start(this.PASSWORD).addClass('full-width-input').end()
+          .start(this.PASSWORD).end()
```
Comment (kgrgreer) [inline src/foam/nanos/auth/SignInView.js]: Here you can also just use .add() instead of start().end().

File: src/foam/u2/view/PasswordView.js [hunk 5]
```diff
@@ -18,7 +18,85 @@
...
+          type: this.type
+        }, this.inputElement$).
+          addClass('input-field').addClass('full-width-input-password').
+        end().
+        start('img').show(this.passwordIcon$).addClass('input-image').
+          attr('src', this.visibilityIcon$).on('click', this.visible).
+        end().
+      end();
+
+      this.inputElement.attrSlot(
```
Comment (kgrgreer) [inline src/foam/u2/view/PasswordView.js]: You don't need to do this. the start('input') is mapped to a foam.u2.tag.Input element which already supports this logic. Just set it's onKey property and then bind to its data$.

File: src/foam/u2/view/PasswordView.js [hunk 6]
```diff
@@ -18,7 +18,77 @@
 foam.CLASS({
   package: 'foam.u2.view',
   name: 'PasswordView',
-  extends: 'foam.u2.tag.Input',
+  extends: 'foam.u2.View',
 
-  properties: [ [ 'type', 'password' ] ]
```
Comment (TW80000) [inline src/foam/u2/view/PasswordView.js:23]: We still need this for the `type` attribute on the HTML input element to be set correctly.

```JavaScript
{
  class: 'String',
  name: 'type',
  value: 'password'
}
```

**Review by TW80000 — CHANGES_REQUESTED** [review]

**Review by TW80000 — APPROVED** [review]

**Pull request merged**